Pricing Grid: Split SFTP/SSH and Git into separate features

sdnunca opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to 3688-gh-Automattic/martech

## Proposed Changes

Splits the `FEATURE_DEV_TOOLS` feature into 2 lines/2 separate features: `FEATURE_DEV_TOOLS_SSH`, `FEATURE_DEV_TOOLS_GIT`

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

Part of an effort to address bugs and inconsistencies in the pricing grids.

## Testing Instructions

* Go to `/setup/onboarding/plans`
* Confirm that the previous feature is shown in the pricing grid:

<img width="786" alt="image" src="https://github.com/user-attachments/assets/490e3975-7eb3-4aa5-94e4-1c68ca23cde5" />

* Open the Comparison table

![image](https://github.com/user-attachments/assets/14c82a0f-70b8-4e96-a75b-7bbab373cc8d)


* Confirm the 2 features are shown separately.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
Pricing Grid: Split SFTP/SSH and Git into separate features

Commit:
Fixes typo

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17420822

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @sdnunca for including a screenshot in the description! This is really helpful for our translators.

claudiucelfilip commented:
Thanks for picking this up!
Works well but I have a couple notes:
* there's some tooltip copy for the separated items in this speadsheet: 379ef-pb
<img width="90" alt="Screenshot 2025-04-25 at 10 04 54" src="https://github.com/user-attachments/assets/3a0be93e-10f4-4a7a-83a8-43e0cdb8b9ea" />
* I'm not sure we want to separate the item in the top plan grid as well (source request p5uIfZ-gt2-p2): @peterschimke can you confirm?

peterschimke commented:
> I'm not sure we want to separate the item in the top plan grid as well

@claudiucelfilip thanks for checking. In the main pricing grid, the developer tools should be listed together as one item. But in the longer 'Compare plans' grid 'SFTP/SSH, WP-CLI' and 'Git commands, and GitHub Deployments' are separate items.

Commit:
Update description copy for the 2 features.

Commit:
Restore the DEV_TOOLS feature as it's still needed in the main pricing grid.

Commit:
Fix test import order

Commit:
Add missing FEATURE_DEV_TOOLS

sdnunca commented:
Thank you both for the feedback. 
I've updated the comparison grid to contain the 2 separate features while keeping the previous feature for the main pricing grid. 

I've also updated the tooltip descriptions to the provided strings.

claudiucelfilip reviewed (approved):
Works as expected.
